### PR TITLE
First cut of doing the geohashing with s2 lib

### PR DESCRIPTION
**Pull request by yifanlisc**
This PR does the followings:

* Creates some unit tests for s2 lib (largely by importing part of the existing unit tests from Google S2)
* Fixes the build.gradle to make it work with the test. However, for now please use "-PskipTest" as the current version of Gradle does not work with Java 8 on task test. 
* Check in the jar lib

### To do

* Upgrade the Gradle to fix the issue on test
* Deploy the jar lib to Artifactory

### CODE REVIEW

**Comment by yifanlisc** [comment]
@sandeepshresthasc , could you take a quick look (primarily on the changes I made on the build.gradle files)? Thanks!

File: aggregator/build.gradle [hunk 1]
```diff
@@ -27,6 +27,8 @@ dependencies {
         exclude group: "org.slf4j", module: "slf4j-jdk14"
     }
 
+    compile files('../libs/s2-geometry-java.jar')
```
Comment (sandeepshresthasc) [inline aggregator/build.gradle]: looks ok now, but check it in artifactory, maybe mustafa can help

**Comment by sandeepshresthasc** [comment]
lgtm other but check in the s2 library to artifactory

File: aggregator/build.gradle (hunk 1 above)
Comment (yifanlisc) [inline aggregator/build.gradle]: Anthony is helping with that now.

**Comment by yifanlisc** [comment]
Squash merge commit b57b783be0c37ee37b7bc8f678d6fc6d9a20db22 created by [Whaler](https://chrome.google.com/webstore/detail/whaler/kjfaikbmbbkbnjjeiambmjchclpfkedf). :whale:

**Pull request closed without merging**